### ISSUE TITLE
[PROMOTION] @SuperBuilder - promote out of experimental

### ISSUE THREAD

**Issue by boonware** [projectlombok/lombok#2209]
I have been using the `@SuperBuilder` annotation for quite some time now yet it is still marked as experimental. In my opinion this is an essential feature when using builders in a class hierarchy. What is the current outlook for moving this out of the experimental stage and into mainstream?

**Comment by janrieke** [comment]
It's the youngest Lombok feature, and it is not even implemented completely, e.g. it's missing some customization possibilities. IMO there are other experimental features that deserve a promotion more.

**Comment by rzwitserloot** [comment]
Yeah this one will remain in experimental for quite a while.

Note that experimental really does not mean 'its going to disappear any moment now'. It just means: At least one of these things applies, possibly all of them:

- We will fix bugs and support this feature, but... not at the same level of priority as non-experimental features
- We might change how it works with significantly less hesitation vs. making breaking changes to non-experimental features
- We are unsure if the feature is a good idea in the first place.

For this one, primarily it's the support aspect of it. We have no intention of dropping the feature and no intention of changing how it works significantly. But it IS a complex one, and we definitely test it less thoroughly right now. Hence, it'll stay experimental for the foreseeable future.

**Comment by nightswimmings** [comment]
It's been 2 years now. Is there any chance to get this as stable? It's the most amazing feature right now for us, but the experimental status is a barrier to approve it

**Comment by Belluk** [comment]
Same here. Any update on the status?

**Comment by ZuperKeurZ** [comment]
@rzwitserloot 

Is there anything we can do to help this feature stable?You mention there should be more tests, are you referring to a lack of unit tests?

**Comment by GuanhuiGuan** [comment]
@rzwitserloot It's been almost 4 years now. Could you plz share some insights into the future of this feature?

**Comment by rzwitserloot** [comment]
I'm not sure I get the concerns here. Internal policy that experimental features must not be used? That really feels like a 'doctor, it hurts when I press here!' situation. Just.. stop pressing.

I can discuss moving it over with @rspilker (and @janrieke Who built the feature and so far has done an excellent job keeping up with issues related to it), but that's down primarily to the notion that the amount of issue fallout from it has been lower than we expected – absolutely not because of peer pressure or because we make a habit of shifting popular features into stable _just_ because they are somewhat popular.

**Comment by rzwitserloot** [comment]
Talked about it with @rspilker. It's being promoted with no changes. If we ever have resolution, `@Builder` on its own makes `@SuperBuilder` almost entirely irrelevant. In that case, we will deprecate `@SuperBuilder`, relegate its support status to 'less urgent' (but still fix bugs, just.. well, less urgent), and then remove it maybe 5 years or more, later. Hence, we are now committing to keeping it around even when it becomes obsolete, for 5 years after we mark it as such.

**Comment by trorter** [comment]
@rzwitserloot so how is it going? Was the promotion accepted?

**Comment by greenlearner01** [comment]
Eagerly looking for this to be accepted as a stable feature

**Comment by javier-garcia-munoz** [comment]
What is the current situation for this annotation?

**Comment by noobjun** [comment]
bump? it's been forever, still experimental? i can make donations to accelerate the process if needed ;)

**Comment by Alwin24** [comment]
@rzwitserloot does this mean @SuperBuilder annotation has been moved to stable feature ? The [official documentation](https://projectlombok.org/features/experimental/) shows it to be still under experimental feature.

**Comment by andreibuboi** [comment]
any updates on this?

**Comment by rzwitserloot** [comment]
Talked it over with @rspilker . @janrieke 's support of it has been stellar so far and if it comes to it, Roel and I don't mind ensuring its support. Next release we'll promote it.

**Comment by pvincendeau** [comment]
Any update regarding this promotion of @SuperBuilder to stable @rzwitserloot @rspilker @janrieke ?

**Comment by cristianhoyos66** [comment]
I'm quite interested on this issue. It seems it's been quite long since the last time there was a review to promote **@SuperBuilder** annotation

do you maybe have any update @rzwitserloot @rspilker @janrieke

**Comment by rzwitserloot** [comment]
Java 22 GA release is in march, you have a guarantee lombok will release within a week, and more likely _before_, 22GA. Given that this issue is in the next-version milestone track we won't forget. We'd like to release earlier than that.

**Comment by Kidlike** [comment]
Would I be a party pooper if I said that `@SuperBuilder` seems to be the only annotation in Lombok that breaks [reproducible builds](https://maven.apache.org/guides/mini/guide-reproducible-builds.html)? :disappointed: 

I haven't managed to pinpoint exactly when and what triggers the problem though... But in our ~100k LoC project, the only classes that generate a different checksum on every build, are the ones annotated with `@SuperBuilder` (some of them are ok though... weirdly enough)

**Comment by janrieke** [comment]
No, I have no idea what could cause that. I need a minimal reproducible example to investigate it further. Please open a new issue for that topic.

**Comment by janrieke** [comment]
Too bad. However I think you should create a new issue with what you found out so far.

**Comment by Rvinx** [comment]
Any update about this?

**Comment by Rvinx** [comment]
Any update regarding this promotion of @SuperBuilder to stable @rzwitserloot @rspilker  ?

**Comment by Junior-Stalin** [comment]
One more year: Any update regarding this promotion of @SuperBuilder  ?

**Comment by jzelAdmin2006** [comment]
Are there any updates on this after almost 6 years? I think this annotation is very useful but I'm heavily discouraged from using it when it's considered experimental.

**Comment by Gouvernathor** [comment]
> If we ever have resolution, @Builder on its own makes @SuperBuilder almost entirely irrelevant. In that case, we will deprecate @SuperBuilder, relegate its support status to 'less urgent'...

I don't understand. Does that mean that `@Builder` will get the `@SuperBuilder` features integrated into it ? Or does it mean that a `@SuperBuilder` will exist (as a stable thing) alongside `@Builder` ? And what does "resolution" mean in that context ?

Also, is the reason why it was not actually promoted in recent releases, the checksum-reproducibility issue mentioned above, or is it another reason ?

**Comment by phash** [comment]
could this be non experimental, please?

**Comment by CasVisser-AMIS** [comment]
Are there any updates on this? I'm also able to contribute if that helps

**Comment by rzwitserloot** [comment]
This is going to be in the (imminent) next release.
